cpp/tests/streaming/test_shuffler.cpp

```diff
@@ -276,17 +274,23 @@ TEST_F(BaseStreamingShuffle, extract_any_before_extract) {
         );
 
         size_t parts_extracted = 0;
+        // For this test we need to drain the shuffler, i.e. ensure all insertion
+        // notifications have been received before extracting. This is only bwecause we
+        // sync_wait each individual extract_any_async.s
+        coro::sync_wait(shuffler->drain());
```

Comment:
let's add a method. I think "draining" before extraction is a bit weird. 
```c++
Node ShufflerAsync::wait_for_all_ready_partitions(){
   co_await latch_;
}
```

Reply:
I will rename, but thinking through I want awaiting finished and shutting down the semaphore to be "one" task. Otherwise it's easy to get it wrong and write something like `coro::when_all(shuffle.finished(), shuffle.drain());` which might shut down the semaphore before all notifications have been received (because there might _currently_ be no outstanding notification tasks).